program-test/src/lib.rs

```diff
@@ -463,13 +463,18 @@ impl Default for ProgramTest {
         let prefer_bpf =
             std::env::var("BPF_OUT_DIR").is_ok() || std::env::var("SBF_OUT_DIR").is_ok();
 
+        // deactivate feature `native_program_consume_cu` to continue support existing mock/test
+        // programs that do not consume units.
+        let mut deactivate_feature_set = HashSet::default();
+        deactivate_feature_set.insert(solana_sdk::feature_set::native_programs_consume_cu::id());
+
```

Comment:
Existing mock/test programs do not consume units, the enforcement added above not only breaks all tests under `program-test/tests` and some zk-token-proof tests, it also breaks downstream tests (such as spl). I deactivated this feature for ProgramTest here to continue support existing tests.